src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1651,8 +1651,8 @@ RelationRestrictionPartitionKeyIndex(RelationRestriction *relationRestriction)
 		partitionKeyTargetAttrIndex++;
 
 		if (!targetEntry->resjunk &&
-			IsPartitionColumn(targetExpression, relationPlannerParseQuery) &&
-			IsA(targetExpression, Var))
+			IsA(targetExpression, Var) &&
```

Comment:
This doesn't change anything, just avoids calling `IsPartitionColumn` unless necessary